Backport 5.0:admin: backup/restore;  iau: containers #27388 (#4006)
uyuni-project/uyuni-docs: modules/administration/pages/ssl-certs-imported.adoc

  If you are using a CA chain, all intermediate CAs must also be available.
* An SSL server private key
* An SSL server certificate

> * An SSL database private key
> * An SSL database certificate

Author: ```suggestion
```

Reviewer: Indeed, remove

You can also list multiple hostnames if your environment requires it.
Supported Key types are [literal]``RSA`` and [literal]``EC`` (Elliptic Curve).

> [IMPORTANT]
> ====
> Database SSL certificates require [literal]``reportdb`` and [literal]``db`` as [literal]``Subject Alternative Name``.
> ====
> 

Author: ```suggestion
```

Reviewer: Remove as well

      --ssl-ca-root string            Root CA certificate path
      --ssl-server-cert string        Server certificate path
      --ssl-server-key string         Server key path

>       --ssl-db-cert string            Database certificate path
>       --ssl-db-key string             Database key path

Author: ```suggestion
```

To replace the certificates, you can replace the installed CA certificate with the new CA, and then update the database.

.Procedure: Replacing All Existing Certificates

> . On the {productname} container host, at the command prompt, recreate podman certificate secrets:

Author: ```suggestion
. On the {productname} container host, at the command prompt, temporarily copy the certificate files to the container:
```

+

[source,shell]
----

> podman secret create --replace uyuni-ca $path_to_ca_certificate
> podman secret create --replace uyuni-db-ca $path_to_database_ca_certificate
> podman secret create --replace uyuni-cert $path_to_server_certificate
> podman secret create --replace uyuni-key $path_to_server_key
> podman secret create --replace uyuni-db-cert $path_to_database_certificate
> podman secret create --replace uyuni-db-key $path_to_database_key

Author: ```suggestion
for f in <Root_CA_Certificate> <Server_Cert_File> <Server_Key_File>; do
  mgrctl cp $f server:/tmp
done
----

. On the {productname} container host, call the following command to run [command]``mgr-ssl-cert-setup`` inside the container providing certificates as parameters:

+

[source,shell]
----
mgrctl exec -ti -- mgr-ssl-cert-setup --root-ca-file=/tmp/<Root_CA_Certificate> \
  --server-cert-file=/tmp/<Server_Cert_File> --server-key-file=/tmp/<Server_Key_File>
----

. Remove the temporarily copied files from the container:

+

[source,shell]
----
mgrctl exec -ti -- rm /tmp/<Root_CA_Certificate> /tmp/<Server_Cert_File> \
  /tmp/<Server_Key_File>
```

mgradm restart
----

> 

Author: ```suggestion

Intermediate CAs can either be available in the file which is specified with `--root-ca-file` or specified as extra options with `--intermediate-ca-file`.
The `--intermediate-ca-file` option can be specified multiple times.
The [command]``mgr-ssl-cert-setup`` command tests if provided files are valid and suitable for the requested use case.
The command then copies or deploys the files to all the places where they are needed.

```
